analyses/mutect2-vs-strelka2/01-set-up.Rmd

```diff
@@ -0,0 +1,476 @@
+transversions and is recommended by the [maftools vignette](https://bioconductor.org/packages/devel/bioc/vignettes/maftools/inst/doc/maftools.html#74_transition_and_transversions) as a method of obtaining an overall base 
+change summary. 
+
+#### Strelka2 transition/transversion plot.
+
+```{r}
+maftools::plotTiTv(maftools::titv(strelka2, plot = FALSE), plotType = 'box')
+```
+
+#### MuTect2 transition/transversion plot.
+
+```{r}
+maftools::plotTiTv(maftools::titv(mutect2, plot = FALSE), plotType = 'box')
+```
+
+## Compare number of mutations per gene
+
+We will our use our premade function, `correlate_maf_summaries`, to evaluate 
+initial correlations between the algorithms on the gene-level.
+These correlations are shown across variant classifications. 
+
+```{r}
+# Correlate the gene summaries
+correlate_maf_summaries(
+  maf1 = strelka2,
+  maf2 = mutect2,
+  maf1_name = "strelka2",
+  maf2_name = "mutect2",
+  summarize_by = "gene",
+  cor_method = "pearson"
+)
+
+# Save to a PDF
+ggplot2::ggsave(file.path("plots", "gene_summary_cor_strelka-vs-mutect2.pdf"))
+```
+
+## Compare number of mutations per sample
+
+Similar to above, we will use our premade function, `correlate_maf_summaries`, 
+to evaluate correlations between the algorithms on the sample-level.
+These correlations are shown across variant classifications. 
+
+```{r}
+# Correlate the sample summaries
+correlate_maf_summaries(
+  maf1 = strelka2,
+  maf2 = mutect2,
+  maf1_name = "strelka2",
+  maf2_name = "mutect2",
+  summarize_by = "sample",
+  cor_method = "pearson"
+)
+
+# Save to a PDF
+ggplot2::ggsave(file.path("plots", "sample_summary_cor_strelka-vs-mutect2.pdf"))
+```
+
+## Set up new variables
+
+Let's set up Strelka2's variables first. 
+
+```{r}
+# Use the premade function to create our new variables
+strelka2_vaf <- set_up_variables(strelka2)
+
+# Take a look at the new variables:
+strelka2_vaf %>%
+  dplyr::select(vaf, base_change, coding, change, mutation_id)
+```
+
+Now we will do the same for MuTect2.
+
+```{r}
+# Use the premade function to create our new variables
+mutect2_vaf <- set_up_variables(mutect2)
+
+# Take a look at the new variables
+mutect2_vaf %>%
+  dplyr::select(vaf, base_change, coding, change, mutation_id)
+```
+
+## Combine MuTect2 and Strelka2 data.frames into one data.frame
+
+Join these datasets based on the `mutation_id` created by the `set_up_variables`
+function. 
+Save to a TSV file to be used in the subsequent notebook.
+
+```{r}
+# Merge these data.frames together
+vaf_df <- strelka2_vaf %>%
+  dplyr::full_join(mutect2_vaf,
+    by = "mutation_id",
+    suffix = c(".strelka2", ".mutect2")
+  ) %>%
+  # Make a variable that denotes which dataset it is in.
+  dplyr::mutate(dataset = dplyr::case_when(
+    is.na(Allele.mutect2) ~ "strelka2_only",
+    is.na(Allele.strelka2) ~ "mutect2_only",
+    TRUE ~ "both"
+  )) %>%
+  readr::write_tsv(file.path("results", "combined_results.tsv"))
```

Comment:
Since you need to specify `"results"` multiple times, if you set `result_dir <- "results"`, you would only have to change it in one place should the need to change it arise.